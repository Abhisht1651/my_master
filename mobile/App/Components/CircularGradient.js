// @flow
import React from 'react'
import { View, StyleSheet } from 'react-native'
import LinearGradient from 'react-native-linear-gradient'
import PropTypes from 'prop-types'
import styles from './Styles/CircularGradientStyles'

type Props = {
  position?: StyleSheet,
  colors?: PropTypes.array,
  linearGradient?: StyleSheet,
  inside?: StyleSheet,
  source?: PropTypes.object
}

const CircularGradient = (props: Props) => {
  const { position, colors, linearGradient, inside, source } = props
  return (
    <View style={position}>
      <LinearGradient colors={colors} style={[styles.linearGradient, linearGradient]}>
        <View style={[styles.inside, inside]}>{source}</View>
      </LinearGradient>
    </View>
  )
}

CircularGradient.defaultProps = {
  position: StyleSheet.create({}),
  colors: undefined,
  linearGradient: StyleSheet.create({}),
  inside: StyleSheet.create({}),
  source: undefined
}
export default CircularGradient
