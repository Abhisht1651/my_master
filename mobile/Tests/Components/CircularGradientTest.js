import { Text } from 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import CircularGradient from '../../App/Components/CircularGradient'

const color = ['rgba(70, 195, 244, 1)', 'rgba(178, 124, 243, 1)']

test('CircularGradient component renders correctly', () => {
  const tree = renderer.create(<CircularGradient colors={color} />).toJSON()
  expect(tree).toMatchSnapshot()
})

test('CircularGradient component with Component renders correctly', () => {
  const tree = renderer
    .create(<CircularGradient colors={color} source={<Text>Hello</Text>} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
