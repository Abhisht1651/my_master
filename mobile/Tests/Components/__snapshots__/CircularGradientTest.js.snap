// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircularGradient component renders correctly 1`] = `
<View
  style={Object {}}
>
  <BVLinearGradient
    colors={
      Array [
        4282827764,
        4289887475,
      ]
    }
    endPoint={
      Object {
        "x": 0.5,
        "y": 1,
      }
    }
    locations={null}
    startPoint={
      Object {
        "x": 0.5,
        "y": 0,
      }
    }
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 250,
          "height": 500,
          "justifyContent": "center",
          "width": 500,
        },
        Object {},
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 245,
            "height": 490,
            "justifyContent": "center",
            "width": 490,
          },
          Object {},
        ]
      }
    />
  </BVLinearGradient>
</View>
`;

exports[`CircularGradient component with Component renders correctly 1`] = `
<View
  style={Object {}}
>
  <BVLinearGradient
    colors={
      Array [
        4282827764,
        4289887475,
      ]
    }
    endPoint={
      Object {
        "x": 0.5,
        "y": 1,
      }
    }
    locations={null}
    startPoint={
      Object {
        "x": 0.5,
        "y": 0,
      }
    }
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 250,
          "height": 500,
          "justifyContent": "center",
          "width": 500,
        },
        Object {},
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 245,
            "height": 490,
            "justifyContent": "center",
            "width": 490,
          },
          Object {},
        ]
      }
    >
      <Text>
        Hello
      </Text>
    </View>
  </BVLinearGradient>
</View>
`;
